db/db_impl/db_impl.cc

```diff
@@ -1356,13 +1356,21 @@ Status DBImpl::ApplyReplicationLogRecord(ReplicationLogRecord record,
 
         auto current_update_sequence = versions_->GetManifestUpdateSequence();
         uint64_t latest_applied_update_sequence = 0;
+        auto replication_epoch =
+            immutable_db_options_.replication_log_listener
+                ->EpochOfReplicationSequence(replication_sequence);
         for (auto& e : edits) {
           if (!e.HasManifestUpdateSequence()) {
             s = Status::InvalidArgument(
                 "Manifest write doesn't have a ManifestUpdateSequence");
             break;
           }
           latest_applied_update_sequence = e.GetManifestUpdateSequence();
+          auto inferred_epoch_of_mus = versions_->GetReplicationEpochForMUS(latest_applied_update_sequence);
+          if (!inferred_epoch_of_mus || (*inferred_epoch_of_mus != replication_epoch)) {
+            info->diverged_manifest_writes = true;
+            break;
+          }
```

Comment:
I'm pretty sure we only need to do this check if the check on L1374 is true (`e.GetManifestUpdateSequence() <= current_update_sequence`), right?